integration/app_integration_test.go

```diff
@@ -64,12 +64,74 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// TestAppAccessForward tests that requests get forwarded to the target application
-// within a single cluster and trusted cluster.
-func TestAppAccessForward(t *testing.T) {
-	// Create cluster, user, sessions, and credentials package.
+// TestAppAccess runs the full application access integration test suite.
+//
+// It allows to make the entire cluster set up once, instead of per test,
+// which speeds things up significantly.
+func TestAppAccess(t *testing.T) {
 	pack := setup(t)
 
+	t.Run("TestAppAccessForward", func(t *testing.T) {
+		AppAccessForward(t, pack)
+	})
+
+	t.Run("TestAppAccessWebsockets", func(t *testing.T) {
+		AppAccessWebsockets(t, pack)
+	})
+
+	t.Run("TestAppAccessClientCert", func(t *testing.T) {
+		AppAccessClientCert(t, pack)
+	})
+
+	t.Run("TestAppAccessFlush", func(t *testing.T) {
+		AppAccessFlush(t, pack)
+	})
+
+	t.Run("TestAppAccessForwardModes", func(t *testing.T) {
+		AppAccessForwardModes(t, pack)
+	})
+
+	t.Run("TestAppAccessRewriteHeadersRoot", func(t *testing.T) {
+		AppAccessRewriteHeadersRoot(t, pack)
+	})
+
+	t.Run("TestAppAccessRewriteHeadersLeaf", func(t *testing.T) {
+		AppAccessRewriteHeadersLeaf(t, pack)
+	})
+
+	t.Run("TestAppAccessLogout", func(t *testing.T) {
+		AppAccessLogout(t, pack)
+	})
+
+	t.Run("TestAppAccessJWT", func(t *testing.T) {
+		AppAccessJWT(t, pack)
+	})
+
+	t.Run("TestAppAccessNoHeaderOverrides", func(t *testing.T) {
+		AppAccessNoHeaderOverrides(t, pack)
+	})
+
+	t.Run("TestAppAuditEvents", func(t *testing.T) {
+		AppAuditEvents(t, pack)
+	})
+
+	t.Run("TestAppInvalidateAppSessionsOnLogout", func(t *testing.T) {
+		AppInvalidateAppSessionsOnLogout(t, pack)
+	})
+
+	// The previous test invalidated the web session so init it again,
+	// otherwise further tests will be getting 403's.
+	pack.initWebSession(t)
```

Comment:
Could you put this inside a `t.Cleanup()` for the `AppInvalidateAppSessionsOnLogout` test? This way if we reorder things we don't have to remember where to put the reinit.